test/tasks/ods-build-go_test.go

```diff
@@ -68,6 +68,60 @@ func TestTaskODSBuildGo(t *testing.T) {
 					cleanModcache(t, wsDir)
 				},
 			},
+			"build go app with build caching": {
+				WorkspaceDirMapping: map[string]string{"source": "go-sample-app"},
+				PreRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+					ctxt.ODS = tasktesting.SetupGitRepo(t, ctxt.Namespace, wsDir)
+					ctxt.Params = map[string]string{
+						"go-os":              runtime.GOOS,
+						"go-arch":            runtime.GOARCH,
+						"sonar-quality-gate": "true",
+						"cache-build":        "true",
+					}
+				},
+				WantRunSuccess: true,
+				PostRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+
+					checkFilesExist(t, wsDir,
+						"docker/Dockerfile",
+						"docker/app",
+						filepath.Join(pipelinectxt.LintReportsPath, "report.txt"),
+						filepath.Join(pipelinectxt.XUnitReportsPath, "report.xml"),
+						filepath.Join(pipelinectxt.CodeCoveragesPath, "coverage.out"),
+						filepath.Join(pipelinectxt.SonarAnalysisPath, "analysis-report.md"),
+						filepath.Join(pipelinectxt.SonarAnalysisPath, "issues-report.csv"),
+						filepath.Join(pipelinectxt.SonarAnalysisPath, "quality-gate.json"),
+					)
+
+					sonarProject := sonar.ProjectKey(ctxt.ODS, "")
+					checkSonarQualityGate(t, ctxt.Clients.KubernetesClientSet, ctxt.Namespace, sonarProject, true, "OK")
+
+					// This is not available when build skipping as the default is
+					// supplied on the second repeat.
+					// Not sure whether the check is significant in the first place.
+					// wantLogMsg := "No sonar-project.properties present, using default:"
+					// if !strings.Contains(string(ctxt.CollectedLogs), wantLogMsg) {
+					// 	t.Fatalf("Want:\n%s\n\nGot:\n%s", wantLogMsg, string(ctxt.CollectedLogs))
+					// }
+
+					b, _, err := command.Run(wsDir+"/docker/app", []string{})
+					if err != nil {
+						t.Fatal(err)
+					}
+					if string(b) != "Hello World" {
```

Comment:
Due to changes made in #500 the output needs to be validated against "Don't communicate by sharing memory, share memory by communicating." for the tests to pass.

Have a look here:
https://github.com/opendevstack/ods-pipeline/blob/2c085cb014b1591e41e036a90c656729878cea54/test/tasks/ods-build-go_test.go#L62